main.js

```diff
@@ -28,10 +30,27 @@ class NonCacheRewriter {
     } else {
       this.nativeRewriter = new DummyRewriter()
     }
+    if (config?.orchestrion) {
+      const { instrumentations } = yaml.load(config.orchestrion)
+      this.orchestrionModules = new Set(instrumentations.map((i) => i.module_name))
+    }
   }
 
-  rewrite (code, file) {
-    const response = this.nativeRewriter.rewrite(code, file)
+  rewrite (code, file, passes) {
+    let moduleName
+    let moduleVersion
+    if (passes.includes('orchestrion')) {
+      const details = getNameAndVersion(file)
+      moduleName = details.name
+      moduleVersion = details.version
+      if (!this.orchestrionModules.has(moduleName)) {
+        passes.splice(passes.indexOf('orchestrion'), 1)
+      }
+    }
+    if (passes.length === 0) {
```

Comment:
nit: this should be at the top of the function

Reply:
This could be at the top of the function too, but here is mandatory, because `passes` is filtered few lines above, and could go from 1 to 0 items.

PS: Answering because I suggested this code here 😅 
https://github.com/DataDog/dd-wasm-js-rewriter/pull/122/files/9d4f9999c308ded0988aeb52256b8efb952a53fa#r2028871816